### PR TITLE
Use payload_id established by 3Scale

### PR DESCRIPTION
**Pull request by SteveHNH**
3Scale will be assigning a unique ID to each upload and the upload
service will now use that to track the payload through the platform.
This can be used by support to see where the payload is or where it
failed since all apps will identify it the same way.

The old 'hash' field is still available for backward compatibility for
the time being.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -186,25 +185,29 @@ class MQStatus(object):
             logger.error("handle_file(): unable to decode msg as json: {}".format(msg.value))
             continue
 
-        hash_ = data['hash']
+        if data.get('payload_id'):
+            payload_id = data['payload_id']
+        else:
+            payload_id = data.get('hash')
```
Comment (SteveHNH) [inline app.py]: for backward compatibility. If a validator does not update to use `payload_id` then we still want to be able to accept `hash`. We can pull this out once everyone is updated and we don't see this anymore.

File: app.py [hunk 2]
```diff
@@ -48,7 +47,7 @@
 DUMMY_VALUES = {
     'principal': 'default_principal',
     'rh_account': '000001',
-    'hash': 'abcdef123456',
+    'payload_id': '1234567890abcdef',
```
Comment (Glutexo) [inline app.py]: This value is not used anywhere. Currently only _principal_ and _rh_account_ are [used](https://github.com/RedHatInsights/insights-upload/blob/49c34eb30f20186fc7605e0dbbd77bcba89c5bbb/app.py#L326). This replacement is ok in scope of this PR, I’d just consider removing them in a follow-up one.

File: app.py (hunk 1 above)
Comment (Glutexo) [reply app.py]: What if the payload doesn’t contain neither _payload_id_, nor _hash_? It’s still data from an external service. Do we just want pass _None_ to _storage.copy_? Maybe it’d be good to raise an exception instead.

These four lines can be contracted to a following one line. I am not sure whether it’s more readable though.

```python
payload_id = data.get('payload_id', data.get('hash'))
```

Even if not, I’d find the following version more idiomatic:

```python
if 'payload_id' in data:
    payload_id = data['payload_id']
else:
    payload_id = data.get('hash')
```

or

```python
payload_id = data['payload_id'] if 'payload_id' in data else data.get('hash')
```

File: app.py (hunk 2 above)
Comment (Glutexo) [inline app.py]: This value is not used anywhere. Currently only _principal_ and _rh_account_ are [used](https://github.com/RedHatInsights/insights-upload/blob/49c34eb30f20186fc7605e0dbbd77bcba89c5bbb/app.py#L326). This replacement is ok in scope of this PR, I’d just consider removing them in a follow-up one.

File: app.py [hunk 3]
```diff
@@ -186,25 +185,29 @@ class MQStatus(object):
...
+        else:
+            payload_id = data.get('hash')
         result = data['validation']
 
-        logger.info('processing message: %s - %s', hash_, result)
+        logger.info('processing message: %s - %s', payload_id, result)
         if result.lower() == 'success':
             url = await IOLoop.current().run_in_executor(
-                None, storage.copy, storage.QUARANTINE, storage.PERM, hash_
+                None, storage.copy, storage.QUARANTINE, storage.PERM, payload_id
```
Comment (Glutexo) [inline app.py]: What if the payload doesn’t contain neither _payload_id_, nor _hash_? It’s still data from an external service. Do we just want pass _None_ to _storage.copy_? Maybe it’d be good to raise an exception or log an error instead. Currently it takes down the whole service.

File: app.py [hunk 4]
```diff
@@ -186,25 +185,29 @@ class MQStatus(object):
...
+                    'msg': {'url': url,
+                            'payload_id': payload_id}
                 }
             )
         elif result.lower() == 'failure':
-            logger.info('%s rejected', hash_)
+            logger.info('%s rejected', payload_id)
             url = await IOLoop.current().run_in_executor(
-                None, storage.copy, storage.QUARANTINE, storage.REJECT, hash_
+                None, storage.copy, storage.QUARANTINE, storage.REJECT, payload_id
```
Comment (Glutexo) [inline app.py]: Dtto.

File: app.py [hunk 5]
```diff
@@ -255,24 +258,24 @@ def get(self):
             filename {str} -- The filename to upload. Should be the tmpfile
                               created by `write_data`
             tracking_id {str} -- The tracking ID sent by the client
-            hash_value {str} -- the uuid for this upload generated by us at time of POST
+            payload_id {str} -- the uuid for this upload generated by us at time of POST
```
Comment (Glutexo) [inline app.py]: It’s not generated by us, but by 3Scale. Moreover, is it really a UUID now, or did it become just an arbitrary unique ID?

File: app.py [hunk 6]
```diff
@@ -287,27 +290,27 @@ def get(self):
...
+    async def process_upload(self, filename, size, tracking_id, payload_id, identity, service):
         """Process the uploaded file we have received.
 
         Arguments:
             filename {str} -- The filename to upload. Should be the tmpfile
                               created by `write_data`
             size {int} -- content-length of the uploaded filename
             tracking_id {str} -- The tracking ID sent by the client
-            hash_value {str} -- the uuid for this upload generated by us at time of POST
+            payload_id {str} -- the uuid for this upload generated by us at time of POST
```
Comment (Glutexo) [inline app.py]: Dtto.

**Comment by SteveHNH** [comment]
@Glutexo all fixed up and ready for your review, sir

File: app.py [hunk 7]
```diff
@@ -185,10 +185,12 @@ class MQStatus(object):
             logger.error("handle_file(): unable to decode msg as json: {}".format(msg.value))
             continue
 
-        if data.get('payload_id'):
-            payload_id = data['payload_id']
-        else:
-            payload_id = data.get('hash')
+        if ('payload_id' or 'hash') not in data:
```
[the remark this answers is not in the document]
Comment (SteveHNH) [reply app.py]: @Glutexo awesome. will fix and then merge :)

File: app.py [hunk 8]
```diff
@@ -185,10 +185,12 @@ class MQStatus(object):
...
+            logger.error("payload_id not in message. Payload not removed from quarantine.")
+            return
```
[the remark this answers is not in the document]
Comment (SteveHNH) [reply app.py]: @Glutexo yeah, we can't really identify it. Maybe I can put the account and principal in the error so we could potentially trace down where it came from. Maybe not the exact machine, but the customer at least

**Comment by Glutexo** [comment]
Verified after your last fix. Confirmed working.

**Pull request merged**